### PR TITLE
special case powers before calling full Gruntz algorithm; part of #10262

### PR DESCRIPTION
**Pull request by rwst**
This fixes part of #10262. Examples where performance is improved:

```
A: l = list(x**i for i in range(1000))
   l.append(O(x**1001))
   Add(*l)
B: from symengine import *
   x=Symbol('x')
   _=series(1/(1-x), x, 0, 100, method='symengine')
C: _=series(sin(x+x**2)*cos(x+x**2), x, 0, 1000, method='symengine')

         with patch           without patch

A          0.33s                  32s
B          0.09s                  1.6s
C          4.7s                  33s
```

You wouldn't want to do B/C without SymEngine.

### CODE REVIEW

File: sympy/series/order.py [hunk 1]
```diff
@@ -368,6 +368,14 @@ def contains(self, expr):
                 common_symbols = expr.variables
             if not common_symbols:
                 return None
+            if (self.expr.is_Pow and self.expr.base.is_Symbol
+                and self.expr.exp.is_positive):
+                if expr.expr.is_Pow and self.expr.base == expr.expr.base:
```
Comment (asmeurer) [inline sympy/series/order.py]: expr.exp?

Comment (asmeurer) [inline sympy/series/order.py]: Or I guess self.expr.exp

Comment (asmeurer) [inline sympy/series/order.py]: Oh I see, `expr` is the other Order.

File: sympy/series/order.py [hunk 2]
```diff
@@ -368,6 +368,14 @@ def contains(self, expr):
...
+                    return not (self.expr.exp-expr.expr.exp).is_positive
```
Comment (asmeurer) [inline sympy/series/order.py]: Is this right if is_positive is None?

Comment (rwst) [inline sympy/series/order.py]: Then this is leaving the Pow/Mul in the Add. OTOH if that is None you can't say it's definitely in the order, I think, so it seems ok. I didn't encounter failed tests.

**Comment by rwst** [comment]
Case C still spends a lot of time in `Order.contains` so this is WIP.

File: sympy/series/order.py (hunk 2 above)
Comment (asmeurer) [inline sympy/series/order.py]: If it's None then `not None` will give `True`.

**Comment by rwst** [comment]
If this passes I would leave it so. The times (updated above) are much better now. The time in C is apparently spent with converting the big rationals. Please review.

**Comment by asmeurer** [comment]
Is the code here covered by test cases?

**Comment by rwst** [comment]
By all means, near every test computing univariate series uses this code now, without error. I don't see a way to write a test that fails where others would not fail.

**Comment by asmeurer** [comment]
Sure, writing a test that would fail is impossible unless you're actually fixing a bug. Is it worth adding some kind of "performance regression test", which is something that runs very fast now but would run very slow (so slow we would notice it) if this behavior were removed? Your case A (or something similar) seems like a good candidate.

**Comment by rwst** [comment]
I have added a benchmark. Or do we have timing testing capabilities?

**Comment by bjodah** [comment]
@rwst you can open a PR to this repo if you want:
https://github.com/sympy/sympy_benchmarks

**Comment by rwst** [comment]
@bjodah Thanks, I didn't know that.

**Comment by asmeurer** [comment]
@rwst I meant an actual test, in the tests file (the benchmarks are not actually run by anyway).

**Comment by rwst** [comment]
OK, I take your word on that it gets noticed if `test_order` takes 30 seconds.

**Comment by rwst** [comment]
This should be it now.

**Pull request merged**